.github/workflows/build.yml

```diff
@@ -1,71 +1,108 @@
     steps:
-    - uses: actions/checkout@v2
-    - run: make check-license
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Check license
+      run: make check-license
+
   generate:
     runs-on: ubuntu-latest
     name: Generate
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-go@v2
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Setup Golang Environment
+      uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
       with:
-        go-version: ${{ env.go-version }}
-    - run: make generate && git diff --exit-code
+        go-version: stable
+
+    - name: Generate
+      run: make generate && git diff --exit-code
+
   lint:
     runs-on: ubuntu-latest
     name: Lint
     steps:
-    - uses: actions/checkout@v2
-    - name: golangci-lint
-      uses: golangci/golangci-lint-action@v2
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Lint Go
+      uses: golangci/golangci-lint-action@1481404843c368bc19ca9406f87d6e0fc97bdcfd # v7.0.0
       with:
         version: latest
-        args: --timeout=5m
+
   build:
     runs-on: ubuntu-latest
     name: Build
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-go@v2
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Setup Golang Environment
+      uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
       with:
-        go-version: ${{ env.go-version }}
-    - run: make build
+        go-version: stable
+
+    - name: Build
+      run: make build
+
   unit-tests:
     runs-on: ubuntu-latest
     name: Unit tests
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-go@v2
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Setup Golang Environment
+      uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
       with:
-        go-version: ${{ env.go-version }}
-    - run: make test-unit
+        go-version: stable
+
+    - name: Run unit tests
+      run: make test-unit
+
   e2e-tests:
     runs-on: ubuntu-latest
     name: E2E tests
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Setup Golang Environment
+      uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
+      with:
+        go-version: stable
+
     - name: Start kind & create cluster
-      uses: engineerd/setup-kind@v0.5.0
+      uses: engineerd/setup-kind@71e45b960fc8dd50b4aeabf6eb6ef2ca0920b4c1 # v0.6.2
       with:
-        version: ${{ env.kind-version }}
+        version: "v0.27.0"
```

Comment:
It is better to have this at the top, so one doesn't need to scroll through the file to bump the version

Reply:
I reverted it. If we decide to use renovate we can keep that up to date automatically too.